### PR TITLE
Shannon cpp

### PR DESCRIPTION
**Pull request by bx3**
Bioconda requires reviews prior to merging pull-requests (PRs). To facilitate this, once your PR is passing tests and ready to be merged, please add the please review & merge label so other members of the bioconda community can have a look at your PR and either make suggestions or merge it.

[x ] I have read the guidelines for bioconda recipes.
  This PR adds a new recipe.
[x ] AFAIK, this recipe is directly relevant to the biological sciences (otherwise, please submit to the more general purpose conda-forge channel).
  This PR updates an existing recipe.
[ x] This PR does something else (explain below).
It is a C++ implementation to a denovo RNA assembler, called Shannon, which already has its python recipe in the bioconda.

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

@BiocondaBot please update will cause the BiocondaBot to merge the master branch into a PR
@BiocondaBot please add label will add the please review & merge label.
@BiocondaBot please fetch artifacts will post links to packages and docker containers built by the CI system. You can use this to test packages locally before merging.
For members of the Bioconda project, the following command is also available:

@BiocondaBot please merge will cause packages/containers to be uploaded and a PR merged. Someone must approve a PR first! This has the benefit of not wasting CI build time required by manually merging PRs.
If you have questions, please post them in gitter or ping @bioconda/core in a comment (if you are not able to directly ping @bioconda/core then the bot will repost your comment and enable pinging).

### CODE REVIEW

File: recipes/shannon_cpp/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+{% set version = "0.4.0" %}
+{% set base = "https://github.com/bx3/shannon_cpp/releases/download" %}
+{% set sha256linux = "b9e70b6e64a7de64c91b83d4dfa781b149f5c121fb05cb8692fd53e22a448e0d" %}
+
+package:
+  name: shannon_cpp
+  version: '{{version}}'
+
+source:
+  - url: '{{base}}/v{{version}}/shannon_cpp-Linux64-v{{version}}.tar.gz'   # [linux]
```
Comment (dpryan79) [inline recipes/shannon_cpp/meta.yaml]: Can we simply compile this rather than using a precompiled binary?

Comment (bx3) [reply recipes/shannon_cpp/meta.yaml]: I can work on that, plus I am trying to have one with macOS.

Comment (bx3) [reply recipes/shannon_cpp/meta.yaml]: The code now compiles and should work on MacOs.

**Comment by dpryan79** [comment]
@bioconda-bot please update

**Comment by dpryan79** [comment]
Should we skip OSX builds? It's working under python 2 there, but the tests fail for python3 (at least 3.6).

**Comment by bx3** [comment]
Yes, it is better to skip OS X for now, I think it will take more evaluation to locate the bugs in OS X (some with Rcorrector and some with my build). Relating to the python version, are you suggesting I should constraint the python version in meta.yaml? is python3 not working in neither Linux nor MacOS. 

Update: I replaced the software binary with components to build the software. I remove parts of Linux headers in the code for OS X compatibility, but it is an ongoing task.

thanks,
Bowen

**Comment by dpryan79** [comment]
OK, in the interim we can just skip OSX.

**Comment by dpryan79** [comment]
@bioconda-bot please merge

**Pull request merged**